pkg/ebpf/compiler/compiler.go

```diff
@@ -147,7 +72,44 @@ func NewEBPFCompiler(headerDirs []string, verbose bool) (*EBPFCompiler, error) {
 			fmt.Sprintf("-isystem%s/include/generated/uapi", d),
 		)
 	}
-	ebpfCompiler.kernelCflags = cflags
+	cflags = append(cflags, fmt.Sprintf("-isystem%s", embeddedIncludePath))
+	cflags = append(cflags, "-c", "-x", "c", "-o", "-", "-")
+
+	var clangOut, clangErr, llcOut, llcErr bytes.Buffer
+
+	clangCtx, clangCancel := context.WithTimeout(context.Background(), compilationStepTimeout)
+	defer clangCancel()
+
+	compileToBC := exec.CommandContext(clangCtx, clangBinPath, cflags...)
+	compileToBC.Stdin = in
+	compileToBC.Stdout = &clangOut
+	compileToBC.Stderr = &clangErr
+
+	err := compileToBC.Run()
 
-	return ebpfCompiler, nil
+	if err != nil {
+		if clangCtx.Err() == context.DeadlineExceeded {
+			return fmt.Errorf("error compiling asset to bytecode: operation timed out")
+		}
+
+		return fmt.Errorf("error compiling asset to bytecode: %s", clangErr.String())
+	}
+
+	llcCtx, llcCancel := context.WithTimeout(context.Background(), compilationStepTimeout)
+	defer llcCancel()
+
+	bcToObj := exec.CommandContext(llcCtx, llcBinPath, "-march=bpf", "-filetype=obj", "-o", outputFile, "-")
+	bcToObj.Stdin = &clangOut
+	bcToObj.Stdout = &llcOut
```

Comment:
If we don't actually care about `stdout` for error reporting, then this can be `nil` instead.